From d2acbf2b717a22764392f7e84208d343fcc36638 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Wed, 15 Feb 2012 21:59:17 +0100
Subject: [PATCH 2/3] Move the padsp helper lib into a private library

libpulsedsp is only used as a LD_PRELOAD library
for the padsp wrapper. So no need to have it in
a public library
---
 src/Makefile.am    |   13 +++++++-
 src/utils/padsp    |   86 ----------------------------------------------------
 src/utils/padsp.in |   86 ++++++++++++++++++++++++++++++++++++++++++++++++++++
 3 files changed, 97 insertions(+), 88 deletions(-)
 delete mode 100755 src/utils/padsp
 create mode 100644 src/utils/padsp.in

diff --git a/src/Makefile.am b/src/Makefile.am
index 1084134..15045df 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -86,7 +86,7 @@ EXTRA_DIST = \
 		daemon/esdcompat.in \
 		daemon/start-pulseaudio-x11.in \
 		daemon/start-pulseaudio-kde.in \
-		utils/padsp \
+		utils/padsp.in \
 		utils/qpaeq \
 		modules/module-defs.h.m4 \
 		daemon/pulseaudio.desktop.in \
@@ -756,8 +756,17 @@ libpulse_mainloop_glib_la_LDFLAGS = $(AM_LDFLAGS) $(VERSIONING_LDFLAGS) -version
 ###################################
 
 if HAVE_OSS_WRAPPER
-lib_LTLIBRARIES += libpulsedsp.la
+pkglib_LTLIBRARIES += libpulsedsp.la
 bin_SCRIPTS += utils/padsp
+
+edit = @sed \
+       -e "s|@pkglibdir[@]|$(pkglibdir)|g"
+
+utils/padsp: utils/padsp.in
+	$(edit) $< >$@
+
+CLEANFILES += utils/padsp
+
 endif
 
 libpulsedsp_la_SOURCES = utils/padsp.c
diff --git a/src/utils/padsp b/src/utils/padsp
deleted file mode 100755
index 4fe175c..0000000
--- a/src/utils/padsp
+++ /dev/null
@@ -1,86 +0,0 @@
-#!/bin/sh
-
-# This file is part of PulseAudio.
-#
-# Copyright 2006 Lennart Poettering
-# Copyright 2006 Pierre Ossman <ossman@cendio.se> for Cendio AB
-#
-# PulseAudio is free software; you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License as published by
-# the Free Software Foundation; either version 2 of the License, or
-# (at your option) any later version.
-#
-# PulseAudio is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
-# General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public License
-# along with PulseAudio; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
-# USA.
-
-while getopts 'hs:n:m:MSDd' param ; do
-	case $param in
-		s)
-			PULSE_SERVER="$OPTARG"
-			export PULSE_SERVER
-			;;
-		n)
-			PADSP_CLIENT_NAME="$OPTARG"
-			export PADSP_CLIENT_NAME
-			;;
-		m)
-			PADSP_STREAM_NAME="$OPTARG"
-			export PADSP_STREAM_NAME
-			;;
-		M)
-			PADSP_NO_MIXER=1
-			export PADSP_NO_MIXER
-			;;
-		S)
-			PADSP_NO_SNDSTAT=1
-			export PADSP_NO_SNDSTAT
-			;;
-		D)
-			PADSP_NO_DSP=1
-			export PADSP_NO_DSP
-			;;
-		d)
-			if [ x"$PADSP_DEBUG" = x ]; then
-				PADSP_DEBUG=1
-			else
-				PADSP_DEBUG=$(( $PADSP_DEBUG + 1 ))
-			fi
-			export PADSP_DEBUG
-			;;
-		*)
-			echo "$0 - redirect OSS audio devices to PulseAudio"
-			echo " "
-			echo "$0 [options] application [arguments]"
-			echo " "
-			echo "options:"
-			echo "	-h                  show brief help"
-			echo "	-s <host>[:<port>]  contact a specific PulseAudio server"
-			echo "	-n <name>           client name to report to the server"
-			echo "	-m <name>           stream name to report to the server"
-			echo "	-M                  disable /dev/mixer emulation"
-			echo "	-S                  disable /dev/sndstat emulation"
-			echo "	-D                  disable /dev/dsp emulation"
-			echo "	-d                  enable debug output"
-			exit 0
-			;;
-	esac
-done
-
-shift $(( $OPTIND - 1 ))
-
-if [ x"$LD_PRELOAD" = x ] ; then
-   LD_PRELOAD="libpulsedsp.so"
-else
-   LD_PRELOAD="$LD_PRELOAD libpulsedsp.so"
-fi
-
-export LD_PRELOAD
-
-exec "$@"
diff --git a/src/utils/padsp.in b/src/utils/padsp.in
new file mode 100644
index 0000000..4ecce8c
--- /dev/null
+++ b/src/utils/padsp.in
@@ -0,0 +1,86 @@
+#!/bin/sh
+
+# This file is part of PulseAudio.
+#
+# Copyright 2006 Lennart Poettering
+# Copyright 2006 Pierre Ossman <ossman@cendio.se> for Cendio AB
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+# USA.
+
+while getopts 'hs:n:m:MSDd' param ; do
+	case $param in
+		s)
+			PULSE_SERVER="$OPTARG"
+			export PULSE_SERVER
+			;;
+		n)
+			PADSP_CLIENT_NAME="$OPTARG"
+			export PADSP_CLIENT_NAME
+			;;
+		m)
+			PADSP_STREAM_NAME="$OPTARG"
+			export PADSP_STREAM_NAME
+			;;
+		M)
+			PADSP_NO_MIXER=1
+			export PADSP_NO_MIXER
+			;;
+		S)
+			PADSP_NO_SNDSTAT=1
+			export PADSP_NO_SNDSTAT
+			;;
+		D)
+			PADSP_NO_DSP=1
+			export PADSP_NO_DSP
+			;;
+		d)
+			if [ x"$PADSP_DEBUG" = x ]; then
+				PADSP_DEBUG=1
+			else
+				PADSP_DEBUG=$(( $PADSP_DEBUG + 1 ))
+			fi
+			export PADSP_DEBUG
+			;;
+		*)
+			echo "$0 - redirect OSS audio devices to PulseAudio"
+			echo " "
+			echo "$0 [options] application [arguments]"
+			echo " "
+			echo "options:"
+			echo "	-h                  show brief help"
+			echo "	-s <host>[:<port>]  contact a specific PulseAudio server"
+			echo "	-n <name>           client name to report to the server"
+			echo "	-m <name>           stream name to report to the server"
+			echo "	-M                  disable /dev/mixer emulation"
+			echo "	-S                  disable /dev/sndstat emulation"
+			echo "	-D                  disable /dev/dsp emulation"
+			echo "	-d                  enable debug output"
+			exit 0
+			;;
+	esac
+done
+
+shift $(( $OPTIND - 1 ))
+
+if [ x"$LD_PRELOAD" = x ] ; then
+   LD_PRELOAD="@pkglibdir@/libpulsedsp.so"
+else
+   LD_PRELOAD="$LD_PRELOAD @pkglibdir@/libpulsedsp.so"
+fi
+
+export LD_PRELOAD
+
+exec "$@"
